### DISCUSSION TITLE
Another pagination funny - links incorrect

### DISCUSSION THREAD

**Discussion by scottjo** [livewire/livewire#7708] [1. Help] [answered]
Good morning all

So I have a paginated data on a page /cars and looking at the links in the pagination I see /cars?page=2 /cars?page=3 etc. I have a separate 'order by' widget that when changed, saves the selected order by in the session and dispatches an order-updated message that the 'catalogue' is listening out for. 

When the 'catalogue' hears that message, it updates its internal order_by details and then calls 

`$this->resetPage();`

Sure enough the, the first page of the catalogue appears in the correct sequence. However, the paginator itself seems a little odd - the links now read: (Note I've removed the classes from the snippet for brevity

`<a href="https://gladys.test/livewire/update?page=2" class='...' aria-label="Go to page 2">2</a>`

NOTE the destination is now **/livewire/update** 

Refresh the page and the pagination returns to /cars?page=2 etc.

I've surrounded the links call in an empty div, tried surrounding it with a separate wire:key'd div to no avail.

Any ideas?

Thanks in advance

Jon

**Comment by joshhanley** [comment]
@scottjo Make sure your using the `WithPagination` trait in your component.

**Reply by scottjo** [reply]
Confirmed

`use Livewire\WithPagination;

class Catalogue extends Component
{
    use WithPagination;`

**Reply by scottjo** [reply]
Its odd - I have a another widget that alters the filtering of the dataset - works the same way - the catalogue responds to a filter-changed message and that works well - just the ordering of the dataset - so the data set number stays the same but the sequence changes.

**Reply by joshhanley** [reply]
@scottjo Hmm can you share your component class and view? 

When you paste code into github, you need to surround it in triple back ticks and then specify the file type like php or blade.

For example
\```php
\\ PHP code here
\```

\```blade
\\ HTML and blade code here
\```

**Reply by scottjo** [reply]
(No taking the pee :-))

**Reply by joshhanley** [reply, marked as answer]
@scottjo ah so it looks like you are using a custom pagination view. But Livewire's pagination views are different to Laravel's. So if you want your custom pagination view to work with Livewire, you will need to publish Livewire's pagination view using `php artisan livewire:publish --pagination` and then customise that (or make a copy of it and customise it) for your farm pagination.

**Reply by scottjo** [reply]
Ahh ok - let me try that - Thanks @joshhanley

**Reply by joshhanley** [reply]
The main difference between Laravel and Livewire pagination is, Laravel's pagination uses `<a>` tags, where are Livewire's uses `<button>` tags with `wire:click` on them, to ensure that Livewire loads the new pagination page instead of the full page refreshing and loosing all the Livewire state.

**Reply by scottjo** [reply]
Cool - thanks - that makes sense - thanks for your help. Just published the Livewire views and that works as expected. As usual it's a PICNIC error. Thanks again.

**Reply by joshhanley** [reply]
@scottjo no worries! 🙂
